app/javascript/components/MainContainer/MainContainer.js

```diff
@@ -3,79 +3,88 @@ import RestaurantsList from "./RestaurantsList/RestaurantsList";
 import Filters from "./Filters/Filters"
 import Map from "./Map/Map"
 import './main_container.css'
+import { connect } from "react-redux";
+import { restListResponse, restChangeFilter } from "../actions/restaurantsActions";
+
 const axios = require('axios');
 
 
+@connect((store) => {
+    return {
+        restaurants: store.restaurantsList
+    };
+})
 class MainContainer extends Component {
     constructor(props) {
         super(props);
-        this.state = {
-            rating_filter: 'all',
-            cuisine_filter: 'all',
-            ten_bis_filter: 'all',
-            search_text: '',
-            restaurants: [],
-            filtered_restaurants: [],
-            cuisines: []
-        };
-
-        this.onChangeFilter = this.onChangeFilter.bind(this);
-        this.onChangeSearch = this.onChangeSearch.bind(this);
-        this.filterRestaurants = this.filterRestaurants.bind(this);
-
     }
```

Comment:
No need to keep an empty constructor (one with only `super(props)` counts as empty).